src/HotChocolate/Spatial/src/Types.Spatial/GeoJSONPolygonInput.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using HotChocolate.Configuration;
+using HotChocolate.Language;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors.Definitions;
+using NetTopologySuite.Geometries;
+
+namespace Types.Spatial
+{
+    public class GeoJSONPolygonInput : InputObjectType<Polygon>
+    {
+        private const string _typeFieldName = "type";
+        private const string _coordinatesFieldName = "coordinates";
+        private IInputField _typeField = default!;
+        private IInputField _coordinatesField = default!;
+
+        public GeoJSONPolygonInput() { }
+
+        protected override void Configure(IInputObjectTypeDescriptor<Polygon> descriptor)
+        {
+            descriptor.BindFieldsExplicitly();
+
+            descriptor.Field(_typeFieldName).Type<EnumType<GeoJSONGeometryType>>();
+
+            descriptor.Field(_coordinatesFieldName).Type<ListType<GeoJSONPositionScalar>>();
+        }
+
+        public override object? ParseLiteral(IValueNode literal)
+        {
+            if (literal is NullValueNode)
+            {
+                return null;
+            }
+
+            if (!(literal is ObjectValueNode obj) || obj.Fields.Count < 2)
+            {
+                throw new InputObjectSerializationException(
+                    "Failed to serialize Polygon. Needs at least type and coordinates fields");
+            }
+
+            IList<Coordinate>? coordinates = null;
+            GeoJSONGeometryType? type = null;
+
+            for (var i = 0; i < obj.Fields.Count; i++)
+            {
+                ObjectFieldNode field = obj.Fields[i];
+
+                switch (field.Name.Value)
+                {
+                    case _coordinatesFieldName:
+                        coordinates =
+                            (IList<Coordinate>)_coordinatesField.Type.ParseLiteral(field.Value);
```

Comment:
Just saw that we are copying this into an array of Coordinate. we could also implement the scalar in a way that it is an array so that we do not need to copy.

Reply:
I asked for help on this in the slack channel. Do you remind responding there?

Reply:
Yes, I just put the comment in so we do not forget